src/oc_erchef/apps/chef_index/src/chef_solr.erl

```diff
@@ -44,22 +107,32 @@ make_query_from_params(ObjType, QueryString, Start, Rows) ->
     FilterQuery = make_fq_type(ObjType),
     %% 'sort' param is ignored and hardcoded because indexing
     %% scheme doesn't support sorting since there is only one field.
-    Sort = "X_CHEF_id_CHEF_X asc",
+    Sort = binary_to_list(id_field()) ++ " asc",
     #chef_solr_query{query_string = check_query(QueryString),
                      filter_query = FilterQuery,
+                     search_provider = search_provider(),
                      start = decode({nonneg_int, "start"}, Start, 0),
                      rows = decode({nonneg_int, "rows"}, Rows, 1000),
                      sort = Sort,
                      index = index_type(ObjType)}.
 
--spec add_org_guid_to_query(#chef_solr_query{}, binary()) ->
-                                   #chef_solr_query{}.
+-spec add_org_guid_to_query(#chef_solr_query{}, binary())
+                           -> #chef_solr_query{}.
 add_org_guid_to_query(Query = #chef_solr_query{filter_query = FilterQuery},
                       OrgGuid) ->
-    Query#chef_solr_query{filter_query = "+" ++
-                              search_db_from_orgid(OrgGuid) ++
-                              " " ++ FilterQuery}.
+    Query#chef_solr_query{filter_query = ?PROVIDER_FUNC(add_org_guid_to_fq,
```

Comment:
Since we  have   `search_provider` in the  `chef_solr_query`,  can we  directly  Provider:add_org_guid_to_fq(...)? 

Same in other places where we're operating off of a `chef_solr_query` record.